qiita_db/metadata_template/sample_template.py

```diff
@@ -189,6 +187,38 @@ def study_id(self):
         """
         return self._id
 
+    @property
+    def columns_restrictions(self):
+        """Gets the dictionary of colums required
+
+        Returns
+        -------
+        dict
+            The dict of restictions
+        """
+        return SAMPLE_TEMPLATE_COLUMNS
+
+    def can_be_updated(self, **kwargs):
+        """Gets if the template can be updated
+
+        Parameters
+        ----------
+        kwargs : ignored
+            Necessary to have in parameters to support other objects.
+
+        Returns
+        -------
+        bool
+            As this is the sample template this returns always True
+
+        Notes
+        -----
+            This always return True as the sample template can always be
```

Comment:
As @antgonza said, that is tested in another step - by definition of update
you can update values of a sample, not the sample itself...
Anyways, updating a sample ID is as simple as updating the sample ID on
this table. Everything else will be updated because we have a cascade on
update. However, the only way of doing such update is going directly to the
SQL.

2015-06-17 9:10 GMT-07:00 Joshua Shorenstein <notifications@github.com>:

> In qiita_db/metadata_template/sample_template.py
> <https://github.com/biocore/qiita/pull/1242#discussion_r32643864>:
>
> > +    def can_be_updated(self, **kwargs):
> > +        """Gets if the template can be updated
> > +
> > +        Parameters
> > +        ----------
> > +        kwargs : ignored
> > +            Necessary to have in parameters to support other objects.
> > +
> > +        Returns
> > +        -------
> > +        bool
> > +            As this is the sample template this returns always True
> > +
> > +        Notes
> > +        -----
> > +            This always return True as the sample template can always be
>
> Thinking about it more, we definitely do NOT what to have that column
> updated using the current code. If we do want to be able to update the
> sample IDs, we also need the updates to propagate to the study->samples
> join table and the other templates as well, which could get interesting.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/1242/files#r32643864>.
>



-- 
Jose Navas